icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/TestMessageFormat.java

```diff
@@ -2122,4 +2124,49 @@ public void TestMessageFormatNumberSkeleton() {
             assertEquals(messagePattern, expected, sb.toString());
         }
     }
+
+    private static void doTheRealDateTimeSkeletonTesting(Date date, Object[][] cases) {
+        for (Object[] cas : cases) {
+            String messagePattern = (String) cas[0];
+            ULocale locale = (ULocale) cas[1];
+            String expected = (String) cas[2];
+
+            MessageFormat msgf = new MessageFormat(messagePattern, locale);
+            StringBuffer sb = new StringBuffer();
+            FieldPosition fpos = new FieldPosition(0);
```

Comment:
I think we typically use bogus field number -1 to indicate "ignore". I also think that we have a named constant for that. Please take a look.

Reply:
Done